Sources/PenguinStructures/UnsafeBufferAccessibleCollection.swift

```diff
@@ -0,0 +1,31 @@
+//******************************************************************************
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/// A collection whose elements can be accessed using `UnsafeBufferPointer`s
+/// and `UnsafeMutableBufferPointer`s.
+public protocol UnsafeBufferAccessibleCollection: Collection {
```

Comment:
Is this not provided for by https://github.com/apple/swift-evolution/blob/master/proposals/0237-contiguous-collection.md?  Please read it and maybe consider reading the evolution thread.

Surely it should refine `MutableCollection` since it supplies `withUnsafeMutableBufferPointer`?

Raise the abstraction level of your naming.  Any collection can in principle be accessed this way (simply copy to an array).  Do you intend this to mean the collection is actually contiguously stored in memory?

Reply:
> Is this not provided for by https://github.com/apple/swift-evolution/blob/master/proposals/0237-contiguous-collection.md? Please read it and maybe consider reading the evolution thread.

Oh thanks, this looks like it will do what I need. I will try it out.